modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2BackChannelLogoutTestCase.java

```diff
@@ -0,0 +1,286 @@
+    private String oidcAppClientId = "";
+    private OauthAdminClient adminClient;
+    private String sessionDataKeyConsent;
+    private String sessionDataKey;
+    private final String CONSENT = "consent";
+    private final String APPROVE = "approve";
+    private final String SCOPE_APPROVAL = "scope-approval";
+    private final String USER_AGENT = "User-Agent";
+    private final String username;
+    private final String userPassword;
+    private final String activeTenant;
+
+    @DataProvider(name = "configProvider")
+    public static Object[][] configProvider() {
+        return new Object[][]{
+                {TestUserMode.SUPER_TENANT_ADMIN},
+                {TestUserMode.TENANT_ADMIN}
+        };
+    }
+
+    @Factory(dataProvider = "configProvider")
+    public OAuth2BackChannelLogoutTestCase(TestUserMode userMode) throws Exception {
+
+        super.init(userMode);
+        AutomationContext context = new AutomationContext("IDENTITY", userMode);
+        this.username = context.getContextTenant().getTenantAdmin().getUserName();
+        this.userPassword = context.getContextTenant().getTenantAdmin().getPassword();
+        this.activeTenant = context.getContextTenant().getDomain();
+    }
+
+    @BeforeClass(alwaysRun = true)
+    public void testInit() throws Exception {
+
+        init();
+        createOIDCApplication();
+        client = HttpClientBuilder.create().setDefaultCookieStore(new BasicCookieStore()).build();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void testCleanUp() throws Exception {
+
+        removeApplications();
+    }
+
+    protected void init() throws Exception {
+
+        super.init();
+        adminClient = new OauthAdminClient(backendURL, sessionCookie);
+    }
+
+    @Test(groups = "wso2.is", description = "Test back channel logout for OIDC.")
+    public void testOIDCLogout() {
+
+        // Login.
+        initiateOIDCRequest(false);
+        performOIDCLogin();
+        performOIDCConsentApproval();
+
+        // OIDC app logout.
+        performOIDCLogout();
+    }
+
+    private void createOIDCApplication() {
+
+        try {
+            OAuthConsumerAppDTO appDTO = new OAuthConsumerAppDTO();
+            appDTO.setApplicationName(OIDC_APP_NAME);
+            appDTO.setCallbackUrl(OAuth2Constant.CALLBACK_URL);
+            appDTO.setOAuthVersion(OAuth2Constant.OAUTH_VERSION_2);
+            appDTO.setGrantTypes(OAuth2Constant.OAUTH2_GRANT_TYPE_AUTHORIZATION_CODE);
+            appDTO.setBackChannelLogoutUrl("http://localhost:" + DEFAULT_TOMCAT_PORT + "/playground2/bclogout");
+
+            adminClient.registerOAuthApplicationData(appDTO);
+            OAuthConsumerAppDTO createdApp = adminClient.getOAuthAppByName(OIDC_APP_NAME);
+            Assert.assertNotNull(createdApp, "Adding OIDC app failed.");
+            oidcAppClientId = createdApp.getOauthConsumerKey();
+        } catch (Exception e) {
+            Assert.fail("Adding OIDC app failed.", e);
```

Comment:
do we need to assert failure here? we can add throw Exception to method signature.

Reply:
Fixed 947bac0